File: MisVersiones/Depanas1/DPanasApp/app/src/main/java/dev/jx/resvegetariano/domain/model/Categoria.kt
package dev.jx.resvegetariano.domain.model

import dev.jx.resvegetariano.data.network.model.CategoriaModel

data class Categoria(
    val id: String,
    val productos: List<Producto>?,
    val nombre: String,
    val fechaRegistro: String
)

fun CategoriaModel.toDomain() =
    Categoria(id, productos?.map { it.toNonPopulatedDomain() }, nombre, fechaRegistro)

fun CategoriaModel.toNonPopulatedDomain() =
    Categoria(id, null, nombre, fechaRegistro)

File: MisVersiones/Depanas1/dpanas-web/models/Categoria.ts
import mongoose from 'mongoose';

const CategoriaSchema = new mongoose.Schema({
    productos: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'Producto',
        },
    ],
    nombre: {
        type: String,
        minlength: 3,
        maxlength: 50,
        required: true,
    },
    fechaRegistro: {
        type: Date,
        default: Date.now,
        required: true,
    },
});

export default mongoose.models.Categoria ||
    mongoose.model('Categoria', CategoriaSchema, 'categorias');
